src/libraries/Microsoft.PowerFx.Core/Public/Config/ComposedReadOnlySymbolTable.cs

```diff
@@ -60,18 +60,19 @@ public override FormulaType GetTypeFromSlot(ISymbolSlot slot)
             return slot.Owner.GetTypeFromSlot(slot);
         }
 
+        TexlFunctionSet<TexlFunction> __functions = null;
+
         // Expose the list to aide in intellisense suggestions. 
-        IEnumerable<TexlFunction> INameResolver.Functions
+        TexlFunctionSet<TexlFunction> INameResolver.Functions
         {
             get
             {
-                foreach (INameResolver table in _symbolTables)
+                if (__functions == null)
```

Comment:
>__functions [](http://example.com/codeflow?start=20&length=11)

cache invalidation?

Reply:
Ie, rebase and pass this test https://github.com/microsoft/Power-Fx/pull/886

SymbolTables do have a VersionHash you can use for invalidation.

Reply:
Yes. 
ReadOnly is like IReadOnlyDictionary - the underlying dictionary can still be changed.